libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -2047,7 +2218,50 @@ class MapboxNavigation @VisibleForTesting internal constructor(
         }
     }
 
-    private fun createChildScope() = threadController.getMainScopeAndRootJob().scope
+    private fun runOnNavigationLooper(block: () -> Unit) {
+        if (navigationOptions.looper == Looper.myLooper()) {
+            block()
+        } else {
+            handler.post {
+                runIfNotDestroyed {
+                    block()
+                }
+            }
+        }
+    }
+
+    private fun <T> getResultFromSDKLooperBlocking(block: () -> T): T {
+        return if (navigationOptions.looper == Looper.myLooper()) {
+            block()
+        } else {
+            val latch = CountDownLatch(1)
+            var result: DispatchingResult<T> = DispatchingResult.Empty
+            handler.post {
+                result = try {
+                    val delegationResult2 = runIfNotDestroyed {
+                        block()
+                    }
+                    DispatchingResult.Success(delegationResult2)
+                } catch (t: Throwable) {
+                    DispatchingResult.Error(t)
+                }
+                latch.countDown()
+            }
+            latch.await()
```

Comment:
I can see the increasing number of ANR complaints...
I don't like this way but I don't have a better solution.